tests/Meilisearch.Tests/SearchTests.cs

```diff
@@ -72,6 +72,28 @@ public async Task CustomSearchWithLimit()
             movies.Hits.First().Genre.Should().NotBeEmpty();
         }
 
+        [Fact]
+        public async Task CustomSearchWithPage()
+        {
+            var movies = (PaginatedSearchResult<Movie>) await _basicIndex.SearchAsync<Movie>("man", new SearchQuery { Page = 1 });
+
+            Assert.Equal(1, movies.Page);
+            Assert.Equal(20, movies.HitsPerPage);
+            movies.Hits.First().Id.Should().NotBeEmpty();
+            movies.Hits.First().Name.Should().NotBeEmpty();
+            movies.Hits.First().Genre.Should().NotBeEmpty();
+        }
+
+        [Fact]
+        public async Task CustomSearchWithPageWithoutTypeCast()
+        {
+            var movies = await _basicIndex.SearchAsync<Movie>("man", new SearchQuery { Page = 1 });
+
+            movies.Hits.First().Id.Should().NotBeEmpty();
```

Comment:
Why we didn't check if `movies.Page` exist in this test?

Reply:
It is because it does not exist. If you don't typecast, all the specific fields from the `PaginatedSearchResult` type will not be available. 

It means if you don't want to typecast, it is ok. Your request will still work. The only thing is that some fields will not be available for you.